lib/kafo/hook_context.rb

```diff
@@ -3,7 +3,17 @@
 
 module Kafo
   class HookContext < BaseContext
-    attr_reader :kafo, :logger
+    # @return [Kafo::KafoConfigure]
+    attr_reader :kafo
+
+    # some of hooks won't print any message because logger is not yet configured
+    # configuration of logger depends on application configration (log level etc.)
```

Comment:
```suggestion
    # Hooks in `pre_migrations` and `boot` stages won't print any message because the logger configuration
    # depends on the application configuration, which is not yet available.
```

Might as well improve this a bit while we are here. I'm unsure though if 'available' is the right word here, maybe 'complete' is better.

Reply:
Initialized might be a better word, but really it shouldn't be used in those stages. I've actually been thinking about how we can actually remove them from the context. Perhaps it should raise an exception.

Reply:
This is a bit confusing as to what it means by won't print any messages. Did this mean in the log file? On the console? When using the progress bar? I know this is true for the progress bar output, but its not true for the log file or console based output that comes with verbose.

Reply:
Given this is only moving the existing documentation I'm going to merge it now. There is plenty of room for more improvement.